site/pages/create-disruption.page.tsx

```diff
@@ -65,6 +64,63 @@ const CreateDisruption = ({ inputs }: CreateDisruptionProps): ReactElement => {
         });
     };
 
+    const addValidity = (e: SyntheticEvent) => {
+        e.preventDefault();
+        setValidityClicked(true);
+        if (
+            pageState.inputs["disruption-start-date"] &&
+            pageState.inputs["disruption-start-time"] &&
+            pageState.inputs["disruption-end-date"] &&
+            pageState.inputs["disruption-end-time"]
+        ) {
+            const validityToAdd = {
+                id: pageState.inputs.validity.length + 1,
+                value: `${pageState.inputs["disruption-start-date"]} ${pageState.inputs["disruption-start-time"]} - ${pageState.inputs["disruption-end-date"]} ${pageState.inputs["disruption-end-time"]}`,
+            };
+            setDisruptionPageState({
+                inputs: {
+                    ...pageState.inputs,
+                    validity: [...pageState.inputs.validity, validityToAdd],
+                    "disruption-start-date": "",
+                    "disruption-end-date": "",
+                    "disruption-start-time": "",
+                    "disruption-end-time": "",
+                },
+                errors: pageState.errors,
+            });
+            setValidityClicked(false);
```

Comment:
This line would not be required if the `validityClicked` variable is set to true in the else block

Reply:
It will if you have a case where you have an empty field and it triggers the errors but then you enter valid information without clicking outside (so the onBlur is not triggered). To make the errors go away and reset the state this is needed